src/js/models/panel_model.js

```diff
@@ -299,6 +299,23 @@
             this.save('shapes', rois);
         },
 
+        setPanelRotation(){
+            var rotation = this.get('rotation') || "0"
+            var panelRotationAngle =  (parseInt(rotation) + 90) % 360
+
+            var viewport = this.getViewportAsRect()
+            var width = this.get('width')
+            var height = this.get('height')
+            var offset = (viewport.height - viewport.width)/2
+            var offset_x = viewport.x-offset
+            var offset_y = viewport.y+offset
+
+            this.cropToRoi({'x': offset_x, 'y': offset_y, 'width': viewport.height, 'height': viewport.width, 'rotation': panelRotationAngle})
```

Comment:
It seems to me that you don't need this `cropToRoi` function here?, and removing it fixes the Undo issue.

Reply:
I found it necessary for rectangular big images. If we don't crop and simply update width/height, then the image appears more zoomed-in, but for non-big images, then removing this method works properly (see below). As I already had some weird issues, regarding big images, which you couldn't reproduce, can you test on a rectangular big image if you have the same issue ?

![image](https://github.com/user-attachments/assets/a61d35f9-5816-4c30-b9e6-4b5a705f2c2d)